### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by AHP15**
In this PR, I added the following:

 - Class `Classroom`
 - Class `Book`
 - Class `Rental`
 - Associations between them

Cheers 👍

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (Nemwel-Boniface) [inline book.rb]: - [x] As a book will be having many rentals, kindly ensure that you set both the setter and getter for rental. This should instead be like this: ```attr_accessor :title, :author, :rentals```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
```
Comment (Nemwel-Boniface) [inline book.rb]: - [x] The @rentals needs to be initialized here. Kindly do so ```@rentals = []```

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Book
```
Comment (Nemwel-Boniface) [inline book.rb:3]: - [x] Also, there needs to be an "add_rental``` method that takes in the person and date as the arguments in this class. KIndly include it.

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (Nemwel-Boniface) [inline rental.rb]: - [x] As a rental belongs to a book and also belongs to a person:, kindly ensure that you set both the setter and getter for rental. This should instead be like this: ```attr_accessor :date, :book, :person```

File: rental.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
```
Comment (Nemwel-Boniface) [inline rental.rb]: - [x] You seem to be missing two important aspects to complete the implementation of the belongs_to association correctly. You are missing ```book.rentals << self``` as the rental is adding itself to the @book. Likewise the ````person.rentals << self``` as the rental is adding itself to the @person. KIndly include them.

File: person.rb [hunk 6]
```diff
@@ -38,7 +38,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @parent_permission = parent_permission
   end
 
-  attr_reader :id, :name, :age
+  attr_reader :id, :name, :age, :rentals
```
Comment (Nemwel-Boniface) [inline person.rb]: - [x] You are required to set the getter and setter for the name age and rentals. Kindly change them from an attribute reader to an accessor.
- [x] Also, do not forget to initialize the rentals as well.
- [x] Also, "A method for adding rentals should be present" eg add_rentals in your Person class. Kindly include these to correctly implement the a "A person has many rentals:" association.

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @Abdessittir       :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (mikethreels) [inline book.rb]: - [ ] please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) on how to create this method. it should take the person and date as arguments and the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

File: person.rb [hunk 8]
```diff
@@ -48,6 +50,10 @@ def correct_name
     @name
   end
 
+  def add_rentals(rental)
+    @rentals.push(rental)
+  end
+
```
Comment (mikethreels) [inline person.rb]: - [ ] please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) on how to create this method. it should take the book and date as arguments and the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi @Abdessittir, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- linters are passing

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 7 above)
Comment (AHP15) [reply book.rb]: Why it should take a person and a date as arguments?

File: person.rb (hunk 8 above)
Comment (AHP15) [reply person.rb]: Why it should take a person and a date as arguments?

**Review by m05e5 — APPROVED** [review]
**Hi @Abdessittir 

You did great. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights
✔️ Descriptive PR 
✔️ your linters are all ok

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**